### ISSUE TITLE
@RequestParam empty lists are expanded as singleton lists containing `null`

### ISSUE THREAD

**Issue by darrenfoong** [OpenFeign/feign#1197]
I'm using Spring Cloud OpenFeign, and I have a use case where I need to send a `List` as a `@RequestParam`, for example:

```
@GetMapping("/items")
Page<Item> getItem(@RequestParam List<String> manufacturer);
```

When I send an empty list (e.g. `remoteService.getItem(Collections.emptyList())`), the receiver receives a singleton list with a single `null`.

**Cause**

The issue seems to be in `Expressions.expandIterable()` ([link](https://github.com/OpenFeign/feign/blob/master/core/src/main/java/feign/template/Expressions.java#L132)), which expands an empty list into `null`, which gets returned to `expand()`, whose `StringBuilder` appends a `null`. `StringBuilder` will append a literal `null` to the internal string buffer, resulting in an expansion like `/items?manufacturer=null`.

**Attempted fix**

I forked Feign and checked the above behaviour in `QueryTemplateTest`. In my investigation (https://github.com/OpenFeign/feign/compare/master...darrenfoong:query-template-empty-list), I modified `expandIterable()` and added tests to check the expected behaviour.

**Comment by EzioL** [comment]
I also met this bug. 😂You can set `required = false` to solve the problem first.

Here is appendNull()  code
` private AbstractStringBuilder appendNull() {
        int c = count;
        ensureCapacityInternal(c + 4);
        final char[] value = this.value;
        value[c++] = 'n';
        value[c++] = 'u';
        value[c++] = 'l';
        value[c++] = 'l';
        count = c;
        return this;
    }`

**Comment by kdavisk6** [comment]
@darrenfoong 

This appears to be a bug in our expansion.  The expected behavior here is that the parameter is omitted completely.  Here is a link to our documentation on undefined vs empty

https://github.com/OpenFeign/feign#undefined-vs-empty-values

So, for this to be correct, these are the test cases and their results:

```java
// empty
Collection<String> items = Collections.emptyList();

// /items?manufacturer=

// null collection
Collection<String> items = null;

// /items
```

I'll look over your PR and review.

**Comment by jebeaudet** [comment]
Any update on this? I also think this should be backported to the 10.7 release as this is the release supported by spring boot 2.2/spring cloud hoxton. Thanks

**Comment by kdavisk6** [comment]
Closed via #1200
